include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -206,6 +206,63 @@ MatrixFree<dim, Number, VectorizedArrayType>::get_cell_level_and_index(
 
 
 
+template <int dim, typename Number, typename VectorizedArrayType>
+std::pair<typename DoFHandler<dim>::cell_iterator, unsigned int>
+MatrixFree<dim, Number, VectorizedArrayType>::get_face_iterator(
+  const unsigned int face_batch_number,
+  const unsigned int vector_number,
+  const bool         interior,
+  const unsigned int fe_component) const
+{
+  std::pair<typename DoFHandler<dim>::cell_iterator, unsigned int> result;
+
+  AssertIndexRange(fe_component, dof_handlers.n_dof_handlers);
+  if (interior)
+    {
+      AssertIndexRange(face_batch_number, n_ghost_inner_face_batches());
+    }
+  else
+    {
+      AssertIndexRange(face_batch_number, n_inner_face_batches());
+    }
+
+  AssertIndexRange(vector_number,
+                   n_active_entries_per_face_batch(face_batch_number));
+
+  const DoFHandler<dim> *dofh = nullptr;
+  if (dof_handlers.active_dof_handler == DoFHandlers::usual)
+    {
+      AssertDimension(dof_handlers.dof_handler.size(),
+                      dof_handlers.n_dof_handlers);
+      dofh = dof_handlers.dof_handler[fe_component];
+    }
+  else
+    {
+      Assert(false,
+             ExcMessage("Cannot return DoFHandler<dim>::cell_iterator "
+                        "for underlying DoFHandler!"));
+    }
+
+  const internal::MatrixFreeFunctions::FaceToCellTopology<
+    VectorizedArrayType::n_array_elements>
+    face2cell_info = get_face_info(face_batch_number);
+
+  const unsigned int cell_index =
+    interior ? face2cell_info.cells_interior[vector_number] :
+               face2cell_info.cells_exterior[vector_number];
+
+  std::pair<unsigned int, unsigned int> index = cell_level_index[cell_index];
+
+  result.first = typename DoFHandler<dim>::cell_iterator(
```

Comment:
And I think that here, you could probably even write the code that follows as
```
  return { typename DoFHandler<dim>::cell_iterator(...), 
               interior ? ... : ...};
```
and avoid the `result` variable altogether.